70_Pushover.pm

```diff
@@ -235,80 +245,100 @@ sub Pushover_CGI() {
         # search for existing receipt
         keys %{ $hash->{READINGS} };
         while ( my ( $key, $value ) = each %{ $hash->{READINGS} } ) {
-            $revReadings{ $value->{VAL} } = $1
-              if ( defined( $value->{VAL} ) && $key =~ /^cb_(\d+)$/ );
+            if ( defined( $value->{VAL} ) && $key =~ /^cb_(\d+)$/ ) {
+                $revReadings{ $value->{VAL} } = $1;
+                $revReadings{ $1 } = $1;
+            }
         }
 
         if ( defined( $revReadings{$receipt} ) ) {
             my $rAct      = "cbAct_" . $revReadings{$receipt};
             my $rAck      = "cbAck_" . $revReadings{$receipt};
             my $rAckAt    = "cbAckAt_" . $revReadings{$receipt};
             my $rAckBy    = "cbAckBy_" . $revReadings{$receipt};
+            my $rAckByDev = "cbAckByDev_" . $revReadings{$receipt};
             my $rCancelId = "cbCancelId_" . $revReadings{$receipt};
             my $rDev      = "cbDev_" . $revReadings{$receipt};
 
             return ( "text/plain; charset=utf-8",
                 "NOK " . $receipt . ": invalid argument 'acknowledged'" )
-              if ( !defined( $webArgs->{acknowledged} )
-                || $webArgs->{acknowledged} ne "1" );
+              if ( defined( $webArgs->{acknowledged} )
+                && $webArgs->{acknowledged} !~ /^[01]$/ );
 
             return ( "text/plain; charset=utf-8",
                 "NOK " . $receipt . ": invalid argument 'acknowledged_by'" )
-              if ( !defined( $webArgs->{acknowledged_by} )
-                || $webArgs->{acknowledged_by} ne $hash->{USER_KEY} );
+              if ( defined( $webArgs->{acknowledged_by} )
+                && $webArgs->{acknowledged_by} !~ /^[A-Za-z0-9]{30}$/ );
```

Comment:
2) A message is now only acknowledge after using the "Acknowledge" button within the Pushover notification. Clicking the link for an action will not automatically acknowledge the message.

This needed to be changed since otherwise the operation would fail if the message is not acknowledged as well